pkg/controller/servicebinding/secret_test.go

```diff
@@ -38,17 +36,98 @@ func TestSecretNew(t *testing.T) {
 		f.FakeDynClient(),
 		ns,
 		name,
+		data,
+		secretOwnerReference,
 	)
 
+	t.Run("unique_secret_name", func(t *testing.T) {
+		hashedName, err := HashedResourceNameForData(name, data)
+		assert.NoError(t, err)
+		assert.Contains(t, hashedName, name)
+	})
+
 	t.Run("createOrUpdate", func(t *testing.T) {
-		u, err := s.createOrUpdate(data, secretOwnerReference)
+		u, err := s.createOrUpdate()
 		assert.NoError(t, err)
+		hashedName, err := HashedResourceNameForData(name, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName, u.GetName())
 		assertSecretNamespacedName(t, u, ns, name)
 	})
 
 	t.Run("Get", func(t *testing.T) {
 		u, err := s.get()
 		assert.NoError(t, err)
+		hashedName, err := HashedResourceNameForData(name, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName, u.GetName())
 		assertSecretNamespacedName(t, u, ns, name)
 	})
+
+	t.Run("different_secret_payload_have_different_name", func(t *testing.T) {
```

Comment:
> different payloads are different secrets, since the secrets are immutable. Consider rephrasing the test name to reflect that.

Done

> On top of that, we still need tests that checks what happens when the payload is the same, in particular when keys are sorted differently.

There would never be a case when payload is same but keys are sorted differently.

If `data` is same, `sort.Strings(keys)` will produce same result.
`

https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-0141e5e3383464862fa2299425f3318875159d89221a6cd2f36088f6d06d75b1R43